build/Build.cs

```diff
@@ -0,0 +1,186 @@
+    Verbose = nameof(DotNetVerbosity.Diagnostic))]
+class Build : NukeBuild
+{
+    public static int Main() => Execute<Build>(x => x.Default);
+    //////////////////////////////////////////////////////////////////////
+    // ARGUMENTS
+    //////////////////////////////////////////////////////////////////////
+    [Parameter] readonly string Configuration = "Release";
+    [Parameter] readonly string SigningCertificatePath = "./certificates/OctopusDevelopment.pfx";
+    [Parameter] readonly string SigningCertificatePassword = "Password01!";
+    ///////////////////////////////////////////////////////////////////////////////
+    // GLOBAL VARIABLES
+    ///////////////////////////////////////////////////////////////////////////////
+    AbsolutePath PublishDir => RootDirectory / "publish";
+    AbsolutePath ArtifactsDir => RootDirectory / "artifacts";
+    AbsolutePath LocalPackagesDir => RootDirectory / ".." / "LocalPackages";
+    AbsolutePath OctopusClientFolder => RootDirectory / "source" / "Octopus.Client";
+    
+    [NukeOctoVersion] readonly OctoVersionInfo OctoVersionInfo;
+
+    Target Clean => _ => _
+        .Executes(() =>
+    {
+        EnsureCleanDirectory(ArtifactsDir);
+        EnsureCleanDirectory(PublishDir);
+        (RootDirectory / "source").GlobDirectories("**/bin").ForEach(x => EnsureCleanDirectory(x));
+        (RootDirectory / "source").GlobDirectories("**/obj").ForEach(x => EnsureCleanDirectory(x));
+        (RootDirectory / "source").GlobDirectories("**/TestResults").ForEach(x => EnsureCleanDirectory(x));
+    });
+
+    Target Restore => _ => _
+        .DependsOn(Clean)
+        .Executes(() =>
+        {
+            DotNetRestore(_ => _
+                .SetProjectFile(RootDirectory / "source")
+                .SetVersion(OctoVersionInfo.FullSemVer));
+        });
+
+    Target Compile => _ => _
+        .DependsOn(Restore)
+        .DependsOn(Clean)
+        .Executes(() =>
+    {
+        DotNetBuild(_ => _
+            .SetProjectFile(RootDirectory / "source")
+            .SetConfiguration(Configuration)
+            .SetVersion(OctoVersionInfo.FullSemVer));
+    });
+
+    Target Test => _ => _
+        .DependsOn(Compile)
+        .Executes(() =>
+    {
+        (RootDirectory).GlobFiles("**/**/*.Tests.csproj").ToList().ForEach(testProjectFile =>
+        {
+            DotNetTest(_ => _
+            .SetProjectFile(testProjectFile)
+            .SetConfiguration(Configuration)
+            .SetNoBuild(true));
+        });
+    });
+
+    Target Merge => _ => _
+        .DependsOn(Test)
+        .Executes(() =>
+        {
+        var inputFolder = OctopusClientFolder / "bin" / Configuration / "net452";
+        var outputFolder = OctopusClientFolder / "bin" / Configuration / "net452Merged";
+        EnsureExistingDirectory(outputFolder);
+        
+        var assemblyPaths = System.IO.Directory.EnumerateFiles(inputFolder, "NewtonSoft.Json.dll").Select(f => (AbsolutePath)f);
```

Comment:
Can we just glob here instead? Seems like we're mixing and matching between nuke and native dotnet

Reply:
Good call - I changed: 

```
var assemblyPaths = System.IO.Directory.EnumerateFiles(inputFolder, "NewtonSoft.Json.dll").Select(f => (AbsolutePath)f);
assemblyPaths = assemblyPaths.Concat(System.IO.Directory.EnumerateFiles(inputFolder, "Octodiff.exe").Select(f => (AbsolutePath)f));

var inputAssemblies = new List<string> { $"{inputFolder}/Octopus.Client.dll" };
inputAssemblies.AddRange(assemblyPaths.Select(x => x.ToString()));
```

to

```
var inputAssemblies = 
                inputFolder.GlobFiles("NewtonSoft.Json.dll", "Octodiff.exe", "Octopus.Client.dll");
```